packages/firefly/src/AppRouter.tsx

```diff
@@ -0,0 +1,188 @@
+import { useIsMswStarted } from "@squide/msw";
+import { useIsRouteMatchProtected, useLogger, useRoutes, type Route } from "@squide/react-router";
+import { useAreModulesReady, useAreModulesRegistered } from "@squide/webpack-module-federation";
+import { useEffect, useMemo, useState, type ReactNode } from "react";
+import { Outlet, RouterProvider, createBrowserRouter, useLocation, type RouterProviderProps } from "react-router-dom";
+
+export type OnLoadPublicDataFunction = (signal: AbortSignal) => Promise<unknown>;
+
+export type onLoadProtectedDataFunction = (signal: AbortSignal) => Promise<unknown>;
+
+export type OnCompleteRegistrationFunction = () => Promise<unknown>;
+
+interface RootRouteProps {
+    fallback: ReactNode;
+    onLoadPublicData?: OnLoadPublicDataFunction;
+    onLoadProtectedData?: onLoadProtectedDataFunction;
```

Comment:
```suggestion
    onLoadProtectedData?: OnLoadProtectedDataFunction;
```

Reply:
don't commit as is, the type has to be changed also

Reply:
Oopsy, missed this one!